packages/backend/src/core/EmailService.ts

```diff
@@ -185,33 +192,38 @@ export class EmailService {
 					validateSMTP: false, // 日本だと25ポートが殆どのプロバイダーで塞がれていてタイムアウトになるので
 				});
 			}
-		} else {
-			validated = { valid: true, reason: null };
+		}
+
+		if (!validated.valid) {
+			let reason = validated.reason ?? null;
+			if (reason === 'regex') {
+				reason = 'format';
+			}
+			return {
+				available: false,
+				reason,
+			};
 		}
 
 		const emailDomain: string = emailAddress.split('@')[1];
 		const isBanned = this.utilityService.isBlockedHost(meta.bannedEmailDomains, emailDomain);
 
-		const available = exist === 0 && validated.valid && !isBanned;
+		if (isBanned) {
+			return {
+				available: false,
+				reason: 'banned',
+			};
+		}
 
 		return {
-			available,
-			reason: available ? null :
-			exist !== 0 ? 'used' :
-			isBanned ? 'banned' :
-			validated.reason === 'regex' ? 'format' :
-			validated.reason === 'disposable' ? 'disposable' :
-			validated.reason === 'mx' ? 'mx' :
-			validated.reason === 'smtp' ? 'smtp' :
-			validated.reason === 'network' ? 'network' :
-			validated.reason === 'blacklist' ? 'blacklist' :
-			null,
+			available: true,
+			reason: null,
 		};
 	}
 
 	private async verifyMail(emailAddress: string, verifymailAuthKey: string): Promise<{
 		valid: boolean;
-		reason: 'used' | 'format' | 'disposable' | 'mx' | 'smtp' | null;
+		reason: 'used' | 'format' | 'disposable' | 'mx' | 'smtp' | 'key' | null;
```

Comment:
If the request failed due to the API key, wouldn't it be better to throw an error rather than showing it to the general user?